torch/export/exported_program.py

```diff
@@ -189,19 +176,22 @@ def _override_composite_implicit_decomp(cia_ops_to_callable, safe=True):
     # replace their CompositeImplicitAutograd kernels with NotImplemented.
     # The only current users of this mode are variants of aten::to that we will
     # replace with aten::_to_copy in FunctionalTensorMode.__torch_dispatch__.
+    from torch._decomp import _get_decomp_for_cia
 
     saved_tables = {}
     patched_ops = set()
     for op_overload, decomp_callable in cia_ops_to_callable.items():
         saved_tables[op_overload] = op_overload.py_kernels.copy()
         patched_ops.add(op_overload)
-
         for override_dispatch_key in _AUTOGRAD_ALIAS_BACKEND_KEYS_TO_OVERRIDE:
             if override_dispatch_key not in op_overload.py_kernels:
                 # TODO (tmanlaibaatar)https://github.com/pytorch/pytorch/issues/129430
                 op_overload.py_impl(override_dispatch_key)(
                     autograd_not_implemented(op_overload, deferred_error=True)
                 )
+        # See NOTE: Registering old CIA to Meta kernel
+        # It is important that we cache this before we override py_kernels.
+        orig_cia_callable = _get_decomp_for_cia(op_overload)
```

Comment:
Oh, I guess `_get_decomp_for_cia` already has the logic to:

(1) run the python dispatcher CIA registration if it exists
(2) run the C++ CIA registration if it exists
(3) error if there is no CIA registration

Reply:
Yep it does.